pkg/session_recording/recordings.jsx

```diff
@@ -349,10 +420,29 @@
             cockpit.location.go([recording.id], cockpit.location.options);
         }
 
+        componentDidUpdate() {
+            this.drawSortArrow(this.state.sorting_field);
+        }
+
         render() {
-            let columnTitles = [_("User"), _("Start"), _("End"), _("Duration")];
+            let columnTitles = [
+                (<div className="sort" onClick={this.sortByUser}><span>{_("User")}</span> <div ref="user"
+                    className="sort_icon"></div></div>),
+                (<div className="sort" onClick={this.sortBySince}><span>{_("Start")}</span> <div ref="start"
+                    className="sort_icon"></div></div>),
+                (<div className="sort" onClick={this.sortByUntil}><span>{_("End")}</span> <div ref="end"
+                    className="sort_icon"></div></div>),
+                (<div className="sort" onClick={this.sortByDuration}><span>{_("Duration")}</span> <div ref="duration"
+                    className="sort_icon"></div></div>),
+            ];
             let list = this.props.list;
             let rows = [];
+
+            if (this.state.sorting_field != null) {
+                list = list.map(this.calcDuration);
```

Comment:
I think it is better to just calculate the duration when we load the recording list, in `journalctlIngest'.